aries_vcx_core/src/anoncreds/credx_anoncreds.rs

```diff
@@ -606,10 +606,10 @@ impl BaseAnonCreds for IndyCredxAnonCreds {
         let credential_id = cred_id.map_or(Uuid::new_v4().to_string(), String::from);
 
         let record_value = serde_json::to_string(&credential)?;
-        let tags_json = serde_json::to_string(&tags)?;
+        let tags_json: HashMap<String, String> = serde_json::from_str(&record_value)?;
```

Comment:
should not be serde_jsoning over `record_value` should be over `&tags`

Reply:
let tags_json: HashMap<String, String> = serde_json::from_value(tags)?;

Reply:
I did this on the first place, but it was not working so I was just trying things up!

Reply:
Just speculating, but perhaps when it failed, you were using serde_json::from_str. serde_json::from_value is what we'd use in this situation (as tags is a Value)

Reply:
Alternatively, we could redo the entire construction of tags to be a HashMap from the start, rather than constructing it as a Value and then converting to HashMap. 

However that's not necessarily in scope for this PR, can be fixed up later

Reply:
> Just speculating, but perhaps when it failed, you were using serde_json::from_str. serde_json::from_value is what we'd use in this situation (as tags is a Value)
i actually tried with `from_value()` as soon as I got to know that tags are a value, but as far as I remember it was being converted to a diff type to what was needed!